### DISCUSSION TITLE
Organizing the Quick Tip for Mental Health tips in separate file

### DISCUSSION THREAD

**Discussion by suhaib-lone** [Amna-Hassan04/Serenity-Guide#85] [Ideas]
Hi there! i want to propose a new additional file for organizing the tips in separate json file in Quick Tips for Mental Health section where the user gets random tips on Get Tips button.  Initially i thought of discussing it for Personalized Tips Section but that issue was already taken.

![newco](https://github.com/user-attachments/assets/8bd7c471-d8e8-4943-9fe0-c42c04a6171b)

The json file will look like this:

![jfile](https://github.com/user-attachments/assets/1139f242-198e-48f9-94fb-b1aa5fbd4cf7)

Advantages:
---> Reducing the excess code in the main file.
-->Makes the code look clean.
-->Easy updation of tips without wasting time scrolling through the app.py file.

If you are ok with it i would like to work on this.
Thank You
-Suhaib

**Comment by suhaib-lone** [comment]
@Amna-Hassan04 What I am thinking is a separate json file for keeping those tips. When the user clicks the get tip button, a random tip is generated from the json file and displays it.

Another approach is creating a SQLite Db for it and store the tips in it.
To Do:
--> creating a database table say "tips" and store the tips in it.
--> integrate it with the main app.py file .
--> keeping the database separated from the main file to make it look clean and
it will seem like nothing has changed.

What do you think? If you are suggesting this approach, please do tell so i can work on it.
Thanks!

**Reply by Amna-Hassan04** [reply]
Yes I like the Idea you can start working on this, @suhaib-lone what will be your choice of data base, someone else I think is working on a mongodb data base to connect the feedback section. ping me if you need any help.

**Reply by suhaib-lone** [reply]
@Amna-Hassan04 haven't worked in mongodb. But if you insist I can look into it. But as the project itself is built in python sqlite is a go to choice. But I will go with mongodb if you recommend it but it may take me some time.

I would like to also add that SQLite which is used as sqlite3 as a standard library in python is serverless. Which means there is no overhead cost of maintenance. 

Please do tell me so that I can work accordingly.

**Reply by Amna-Hassan04** [reply]
@suhaib-lone Yeah i understand but since someone has already used mongodb, it would be a good choice to keep the choice of database consistent throughout the project, so take your time but di integrate mongodb. I will give u level2 points if you complete this.

**Reply by suhaib-lone** [reply]
@Amna-Hassan04  Sure, will do that!

**Comment by suhaib-lone** [comment]
@Amna-Hassan04 I have completed integrating MongoDb with the Quick Tips for Mental Health Section
and here are the video and screenshot to check if anything else is needed or to be changed:
![mogointeg](https://github.com/user-attachments/assets/4831e04c-c270-4a27-acf7-24e281676a30)

The video file:

[m2.webm](https://github.com/user-attachments/assets/bc549d21-6ada-4d5b-93c1-eb2ad9e4596e)

**Reply by Amna-Hassan04** [reply]
@suhaib-lone Looks fine suhaib make a pull request. I'll review the code over their.
